enhancements/clusterGroupUpgradeStatus.md

```diff
@@ -0,0 +1,319 @@
+  - cgu-upgrade-complete-policy2-common-pao-sub-policy-placement-kuttl
+  placementRules:
+  - cgu-upgrade-complete-policy1-common-cluster-version-policy-placement-kuttl
+  - cgu-upgrade-complete-policy2-common-pao-sub-policy-placement-kuttl
+  remediationPlan:
+  - - spoke1
+  - - spoke2
+  - - spoke3
+  safeResourceNames:
+    cgu-upgrade-complete-common-cluster-version-policy-config: cgu-upgrade-complete-common-cluster-version-policy-config-kuttl
+    cgu-upgrade-complete-common-pao-sub-policy-config: cgu-upgrade-complete-common-pao-sub-policy-config-kuttl
+    cgu-upgrade-complete-default-subscription-performance-addon-operator: cgu-upgrade-complete-default-subscription-performance-addon-operator-kuttl
+    cgu-upgrade-complete-policy1-common-cluster-version-policy: cgu-upgrade-complete-policy1-common-cluster-versi-kuttl
+    cgu-upgrade-complete-policy1-common-cluster-version-policy-placement: cgu-upgrade-complete-policy1-common-cluster-version-policy-placement-kuttl
+    cgu-upgrade-complete-policy2-common-pao-sub-policy: cgu-upgrade-complete-policy2-common-pao-sub-polic-kuttl
+    cgu-upgrade-complete-policy2-common-pao-sub-policy-placement: cgu-upgrade-complete-policy2-common-pao-sub-policy-placement-kuttl
+  status:
+    currentBatch: 1
+    currentBatchRemediationProgress:
+      spoke1:
+        policyIndex: 0
+        state: InProgress
+  backup: 
+    clusters: 
+    - "spoke2"
+    - "spoke1"
+    - "spoke3"
+    status:
+     - "spoke1": "Succeeded"
+     - "spoke2": "Succeeded"
+     - "spoke3": "Succeeded"
+ precaching: 
+    clusters: 
+    - "spoke2"
+    - "spoke1"
+    - "spoke3"
+    spec:
+      "platformImage": "quay.io/openshift-release-dev/ocp-release@sha256:b9ede044950f73730f00415a6fe8eb1b5afac34def872292fd0f9392c9b483f1"
+    status:
+    - "spoke1": "Active"
+    - "spoke2": "Active"
+    - "spoke3": "Active"
+
+```
+
+This enhancement proposes to change the CGU status APIs to be as the example below (cgu-upgrade-new). The CGU status contains lists of conditions, selected clusters and canary clusters if defined.
+The cluster/canaryCluster list item present the cluster name, policies list and state. 
+The policies list has the managed policies names and its state on the cluster. The policy state has 4 possible state;
+  - **notApplied**: the policy does not apply to enforce remediation 
+  - **nonCompliant**: the policy applied to enforce remediation but it did not get compliant.
+  - **compliant**: the policy applied to enforce remediation and it is compliant.
+  - **timeout**: the policy applied to enforce remediation but it does not become compliant during the timeout limits defined in the remediationStrategy.
+
+The cluster state has 2 possible state;
+  - **complete**: if all the policies has a compliant state on the cluster
+  - **inProgress**: if all the policies are in compliant/nonCompliant or notApplied state.
+  - **failed**: if at least 1 policy has timeout state.
```

Comment:
I prefer `timedout`

Reply:
okay, we can make it timeout. do we will/have any other failure reason ?